optimize_workspace.sh

```diff
@@ -1,19 +1,33 @@
 #!/bin/bash
 set -o errexit -o nounset -o pipefail
-command -v shellcheck > /dev/null && shellcheck "$0"
+command -v shellcheck >/dev/null && shellcheck "$0"
 
 export PATH="$PATH:/root/.cargo/bin"
 
 rustup toolchain list
 cargo --version
 
-optimize_workspace.py
+# Build artifacts
+echo -n "Building artifacts in workspace..."
+/usr/local/bin/build_workspace.py
+echo "done."
 
-# Postprocess artifacts
+echo -n "Optimizing artifacts in workspace..."
+# Start clean
+rm -rf ./artifacts
+mkdir artifacts
+# Optimize and postprocess artifacts
 (
   cd artifacts
-  chmod -x ./*.wasm
-  sha256sum -- *.wasm > checksums.txt
-)
 
-echo "done"
+  for WASM in ../target/wasm32-unknown-unknown/release/*/*.wasm
```

Comment:
We need to ensure that this search only finds results of the current build. Since we mount a volume into target via `--mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \` it can contain anything.

So at least we need to remove all `target/wasm32-unknown-unknown/release/*/*.wasm` before the build step.

Reply:
`cargo clean` at the beginning? Will try

Reply:
This removes all the dependency builds we want to cache. If we use `cargo clean`, we could also not mount the cache volume.

Reply:
Yes, that's what I was thinking. It shouldn't be a problem to have previous artifacts, because they will be rebuilt anyway.

Anyway, I'll add a command to remove all the wasm files at the beginning.